doc/user-guide/push-notifications/Push-notifications.md

```diff
@@ -0,0 +1,178 @@
+## How to set up Push Notifications
+
+Push notifications are the bread and butter of the modern mobile experience, and MongooseIM has
+support for them. When used together with [MongoosePush][], you get out-of-the-box push
+notifications for FCM (**F**irebase **C**loud **M**essaging) and APNS (**A**pple **P**ush
+**N**otification **S**ervice) providers. And it's easy to extend it to any other protocols of your
+choice.
+
+Server side push notification support is fully compliant with [XEP-0357 Push Notifications][XEP-0357],
+which defines several components that need to work together in order to provide clients with working
+push notifications. However, there's just one non-optimal detail required by the aforementioned XEP:
+that push notifications being a PubSub service — [we can do better than that][pubsub-less].
+
+If you're already familiar with the workings of [XEP-0357][], make sure to have a look at
+[our pubsub-less enhancement][pubsub-less].
+
+You might also want to read about the push notification's [client side configuration][client-side].
+
+All push notification mechanisms use [mod_event_pusher_push][] as a backend implementation,
+read [this][mod_event_pusher] to know more about it.
+
+While the whole setup can be incredibly extensible, we see the following straightforward uses of it.
+
+### XEP-0357 compliant with local PubSub
```

Comment:
please arrange the sections in the next order:

1. XEP-0357 compliant with remote PubSub XMPP server
   - note here that the format of the push IQ is can be adjusted via the mod_event_pusher_push custom-made plugin module.
   - also mention that all the provided plugins are generating IQ that that is specific to our own `push PubSub node` implementation
2. XEP-0357 compliant with MIM used as remote PubSub XMPP server
   - state here one more time that we expect push notification IQ to have specific (not declared in XEP) custom fields, e.g. device token. So if someone wants to use with 3rd party XMPP server, he must ensure that this 3rd party server generates push IQ in that specific format. 
3. XEP-0357 compliant when PubSub is configured at the same MIM cluster and IQ is routed internally.
   - MB mention here that this was the most common use-case. but it's senseless and ineffective to generate extra routing, and in the most cases PubSub was the bottleneck of the whole setup. so we came up to an idea of PubSub-less setup.
4. PubSub-less configuration.

Reply:
actually the first and the second point can be replaced with some general XEP overview with an abstract diagram.

Reply:
Done in 818f62403